src/shared/components/mutationTable/column/ClonalColumnFormatter.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import DefaultTooltip from 'shared/components/defaultTooltip/DefaultTooltip';
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import styles from "./mutationType.module.scss";
+import getCanonicalMutationType from "shared/lib/getCanonicalMutationType";
+import {floatValueIsNA} from "shared/lib/NumberUtils";
+
+interface IMutationTypeFormat {
+    label?: string;
+    longName?: string;
+    className: string;
+    mainType: string;
+    priority?: number;
+}
+
+/**
+ * @author Avery Wang
+ */
+export default class ClonalColumnFormatter {
+    /* Determines the display value by using the impact field.
+     *
+     * @param data  column formatter data
+     * @returns {string}    "Clonal" text value
+     */
+    public static getDisplayValue(data:Mutation[]):string {
+        return ClonalColumnFormatter.getClonalValue(data);
+    }
+
+    public static getTooltipValue(data:Mutation[]):string {
+        const ccfMCopiesValue = ClonalColumnFormatter.getCcfMCopiesValue(data);
+        if (floatValueIsNA(ccfMCopiesValue)) {
+            return "FACETS data not available";
+        } 
+        return "CCF: " + ClonalColumnFormatter.getTextValue(ccfMCopiesValue);
+    } 
+            
+    public static getCcfMCopiesUpperValue(data:Mutation[]):number {
+        const ccfMCopiesUpperValue = data[0].ccfMCopiesUpper;
+        return ccfMCopiesUpperValue;
+    }
+
+    public static getCcfMCopiesValue(data:Mutation[]):number {
```

Comment:
if you return null as a number, does it become 0 or remain null? If you are returning a zero here, then ok. Otherwise the test in getTooltipValue() above has the same potential problem (that null will cause false in the test floatValueIsNA(null)